tests/test_logcat.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import io
+import logging
+import os
+import unittest
+
+from llama_stack import logcat
+
+
+class TestLogcat(unittest.TestCase):
+    def setUp(self):
+        self.original_env = os.environ.get("LLAMA_STACK_LOGGING")
+
+        self.log_output = io.StringIO()
+        self._init_logcat()
+
+    def tearDown(self):
+        if self.original_env is not None:
+            os.environ["LLAMA_STACK_LOGGING"] = self.original_env
+        else:
+            os.environ.pop("LLAMA_STACK_LOGGING", None)
+
+    def _init_logcat(self):
+        logcat.init(default_level=logging.DEBUG)
+        self.handler = logging.StreamHandler(self.log_output)
+        self.handler.setFormatter(logging.Formatter("[%(category)s] %(message)s"))
+        logcat._logger.handlers.clear()
+        logcat._logger.addHandler(self.handler)
+
+    def test_basic_logging(self):
+        logcat.info("server", "Info message")
+        logcat.warning("server", "Warning message")
+        logcat.error("server", "Error message")
+
+        output = self.log_output.getvalue()
+        self.assertIn("[server] Info message", output)
+        self.assertIn("[server] Warning message", output)
+        self.assertIn("[server] Error message", output)
+
+    def test_different_categories(self):
+        # Log messages with different categories
+        logcat.info("server", "Server message")
+        logcat.info("inference", "Inference message")
+        logcat.info("router", "Router message")
+
+        output = self.log_output.getvalue()
+        self.assertIn("[server] Server message", output)
+        self.assertIn("[inference] Inference message", output)
+        self.assertIn("[router] Router message", output)
+
+    def test_env_var_control(self):
+        os.environ["LLAMA_STACK_LOGGING"] = "server=debug;inference=warning"
```

Comment:
Wondering what's the difference between having `logcat` + LLAMA_STACK_LOGGING compared with using the logging library. 

E.g. In `remote/inference/fireworks.fireworks.py`
```python
logger = logging.getLogger('inference')
```
+ config with LOG_CONFIG
```python
import logging.config

LOG_CONFIG = {
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
            'formatter': 'simple',
            'level': 'DEBUG'
        },
     },
    'loggers': {
        'inference': {
                'handlers': ['console'],
                'level': 'WARNING',
        }
    }
}
```

Reply:
@yanxi0830 interesting, I did not know about `LOG_CONFIG`. I think I could have used that one. This one feels a bit simpler to use though? unclear. I could have used `LOG_CONFIG` if I had known about it :) l like that there's a static list of categories all logs need to fit into. if people hate this, we can revert :)

Reply:
@ashwinb @yanxi0830 I'm working on a follow-up proposal that uses logConfig too. I should be able to send something shortly. Thanks!